freqtrade/freqai/freqai_interface.py

```diff
@@ -637,6 +659,56 @@ def train_timer(self, do='start'):
                 self.train_time = 0
         return
 
+    def backtest_prediction_exists(
+        self,
+        pair: str,
+        dk: FreqaiDataKitchen,
+        trained_timestamp: int,
+        scanning: bool = False,
+    ) -> bool:
+        """
+        Given a pair and path, check if a backtesting prediction already exists
+        :param pair: pair e.g. BTC/USD
+        :param path: path to prediction
+        :return:
+        :boolean: whether the prediction file exists or not.
+        """
+        if not self.live:
+            prediction_file_name, root_prediction = self.get_backtesting_prediction_file_name(
+                pair, dk, trained_timestamp
+            )
+            path_to_predictionfile = Path(dk.full_path / root_prediction / prediction_file_name)
+
+            file_exists = path_to_predictionfile.is_file()
+            if file_exists and not scanning:
+                logger.info("Found backtesting prediction file at %s", prediction_file_name)
+            elif not scanning:
+                logger.info(
+                    "Could not find backtesting prediction file at %s", prediction_file_name
+                )
+            return file_exists
+        else:
+            return False
+
+    def get_backtesting_prediction_file_name(
+        self, pair: str, dk: FreqaiDataKitchen, trained_timestamp: int
+    ):
+        """
+        Given a pair, path and a trained timestamp,
+        returns the path and name of the predictions file
+        :param pair: pair e.g. BTC/USD
+        :param dk: FreqaiDataKitchen
+        :trained_timestamp: current backtesting timestamp period
+        :return:
+        :str: prediction file name
+        :str: prediction root path
+        """
+        coin, _ = pair.split("/")
+        prediction_base_filename = f"{coin.lower()}_{trained_timestamp}"
+        root_prediction = 'backtesting_predictions'
```

Comment:
Please assign this as `self.backtesting_results_path` to self in the `__init__()` of `class FreqaiDatakitchen` so that you can refer to it anywhere using `dk.backtesting_results_path`

Reply:
Thanks for the feedback. Fixed!